cache/cache_internal_test.go

```diff
@@ -0,0 +1,569 @@
+// +build !windows
```

Comment:
There's no reason `cache` to not work on windows (not tested though) but for some reason the internal tests fail there.

`The process cannot access the file because it is being used by another process.`
This is what the AppVeyor test says. I suspect it's because a file is removed really fast, while it's still in an open state? But why wouldn't the other remotes fail too? Perhaps it's related to the workers (the ones that preload the stream on read) which are still doing a bit of work and conflict with the immediate remove.
It's all theoretical though. A bit hard to see without a windows OS. I'm going to wait for your comment on it, maybe you've seen this before

Reply:
I think I was doing something wrong there when splitting the stream to be cached. I'll check the result of the test now but otherwise I'll need to replicate the dev workspace in a windows vm to see what's up in there

Reply:
Nah, that wasn't. Cache writes aren't even enabled in those cases. Not sure what else could cause this. I'll need to investigate.

Reply:
> The process cannot access the file because it is being used by another process.

You can't do anything to an open file on Windows.  Unix doesn't care - you can delete it, rename it etc.

> I suspect it's because a file is removed really fast, while it's still in an open state?

That is probably correct.

I'm happy to leave tests on windows disabled for the moment if you want - we can make it work in a later commit if you want.

Reply:
Found the bug, it was a race which was visible on Windows only. Because there are workers on separate goroutines opening files they were still doing small pieces of work even after a Close was sent on the reader. Close is now waiting for the workers to close too, not just tell them to stop.

To cut it short, seems Windows' restrictive methods are useful.